### ISSUE TITLE
Error with non-ASCII file names on python2.x

### ISSUE THREAD

**Issue by zimeon** [resync/resync#22]
Problem while trying to use with python2.7:

```
Traceback (most recent call last):
File ".../miniconda2/bin/resync", line 281, in <module>
main()
File ".../miniconda2/bin/resync", line 252, in main
dump=args.resourcedump)
File ".../miniconda2/lib/python2.7/site-packages/resync/client.py", line 637, in write_resource_list
rl.write(basename=outfile)
File ".../miniconda2/lib/python2.7/site-packages/resync/list_base_with_index.py", line 305, in write
s.resources_as_xml(chunk, fh=f)
File ".../miniconda2/lib/python2.7/site-packages/resync/sitemap.py", line 114, in resources_as_xml
tree.write(fh,encoding='UTF-8',xml_declaration=True,method='xml')
File ".../miniconda2/lib/python2.7/xml/etree/ElementTree.py", line 820, in write
serialize(write, self._root, encoding, qnames, namespaces)
File ".../miniconda2/lib/python2.7/xml/etree/ElementTree.py", line 939, in _serialize_xml
_serialize_xml(write, e, encoding, qnames, None)
File ".../miniconda2/lib/python2.7/xml/etree/ElementTree.py", line 939, in _serialize_xml
_serialize_xml(write, e, encoding, qnames, None)
File ".../miniconda2/lib/python2.7/xml/etree/ElementTree.py", line 937, in _serialize_xml
write(_escape_cdata(text, encoding))
File ".../miniconda2/lib/python2.7/xml/etree/ElementTree.py", line 1073, in _escape_cdata
return text.encode(encoding, "xmlcharrefreplace")
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 119: ordinal not in range(128)
```

**Comment by zimeon** [comment]
With test data can now replicate issue on python 2.7, not a problem on python 3.5:

```
(py2)simeon@RottenApple resync (non_ascii_uris)> python --version
Python 2.7.12 :: Continuum Analytics, Inc.
(py2)simeon@RottenApple resync (non_ascii_uris)> resync -r x: tests/testdata/odd_file_names 
Traceback (most recent call last):
  File "/Users/simeon/miniconda3/envs/py2/bin/resync", line 4, in <module>
    __import__('pkg_resources').run_script('resync==1.0.2', 'resync')
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/site-packages/pkg_resources/__init__.py", line 719, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1511, in run_script
    exec(script_code, namespace, namespace)
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/site-packages/resync-1.0.2-py2.7.egg/EGG-INFO/scripts/resync", line 280, in <module>
    
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/site-packages/resync-1.0.2-py2.7.egg/EGG-INFO/scripts/resync", line 248, in main
    
  File "build/bdist.macosx-10.6-x86_64/egg/resync/client.py", line 536, in write_resource_list
  File "build/bdist.macosx-10.6-x86_64/egg/resync/list_base_with_index.py", line 208, in as_xml
  File "build/bdist.macosx-10.6-x86_64/egg/resync/list_base.py", line 133, in as_xml
  File "build/bdist.macosx-10.6-x86_64/egg/resync/sitemap.py", line 127, in resources_as_xml
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/xml/etree/ElementTree.py", line 820, in write
    serialize(write, self._root, encoding, qnames, namespaces)
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/xml/etree/ElementTree.py", line 939, in _serialize_xml
    _serialize_xml(write, e, encoding, qnames, None)
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/xml/etree/ElementTree.py", line 939, in _serialize_xml
    _serialize_xml(write, e, encoding, qnames, None)
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/xml/etree/ElementTree.py", line 937, in _serialize_xml
    write(_escape_cdata(text, encoding))
  File "/Users/simeon/miniconda3/envs/py2/lib/python2.7/xml/etree/ElementTree.py", line 1073, in _escape_cdata
    return text.encode(encoding, "xmlcharrefreplace")
UnicodeDecodeError: 'ascii' codec can't decode byte 0xcc in position 6: ordinal not in range(128)
(py2)simeon@RottenApple resync (non_ascii_uris)> py3
discarding /Users/simeon/miniconda3/envs/py2/bin from PATH
prepending /Users/simeon/miniconda3/envs/py3/bin to PATH
(py3)simeon@RottenApple resync (non_ascii_uris)> resync -r x: tests/testdata/odd_file_names | wc -l
      10
(py3)simeon@RottenApple resync (non_ascii_uris)>
```

**Comment by zimeon** [comment]
Have merged into master 46e154375

**Comment by zimeon** [comment]
Fixed as v1.0.5 and pushed to pypi https://pypi.python.org/pypi?:action=display&name=resync&version=1.0.5
